### ISSUE TITLE
Feature request: Scroll Snap

### ISSUE THREAD

**Issue by becky112358** [linebender/druid#2056]
Thanks for this awesome crate :)

It would be great if the `Scroll` widget had a feature such that when its contents change, it automatically snaps to the "end" of its contents.

I think this idea is mentioned amongst the wishes here: https://github.com/linebender/druid/issues/73

Also I tried to add this feature myself. Although this is not the way that the feature should be implemented, the example may be helpful...: https://github.com/linebender/druid/pull/1798

**Comment by cmyr** [comment]
I agree that something like this is definitely helpful, but there are often subtleties; for instance if the user has scrolled to a specific position, you might not want to automatically scroll them away from that. For the time being, I think you can achieve this using `Command`s and `Controller`s: some controller watches the content view, and then when the size changes it sends a message to a controller wrapping the scroll view, which updates the viewport.

**Comment by becky112358** [comment]
Yes - the Scroll Snap feature should certainly provide an ability for the user to pause the scrolling. Then if they have scrolled to a specific position and don't want to be snapped away, they pause the snapping. See my example. :wink:

I realised another subtlety which I hadn't covered in my previous attempts to create this feature. If you have a `Scroll` and something is added to the middle of the contents, you don't want the `Scroll` to snap to the "bottom".

Hmm... I tried a bunch of stuff with `Command`s, `Controller`s, and local `Widget`s, but nothing works. ☹️ https://github.com/becky112358/druid_scroll_snap . I think your suggestion is my attempt `two controllers`.
